x/finality/keeper/msg_server.go

```diff
@@ -53,9 +53,13 @@ func (ms msgServer) AddFinalitySig(goCtx context.Context, req *types.MsgAddFinal
 		return nil, err
 	}
 
-	fpPK := req.FpBtcPk
-
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	activationHeight, errMod := ms.validateActivationHeight(ctx, req.BlockHeight)
+	if errMod != nil {
+		return nil, errMod.Wrapf("finality block height: %d is lower than activation height %d", req.BlockHeight, activationHeight)
```

Comment:
Nit: The error message should be returned from inside the validation function, for generality.